ui/app/js/controllers/user_edit_controller.js

```diff
@@ -29,29 +34,16 @@ function ($scope, $modalInstance, CSRF, Permissions, users, is_edit, user, permi
       $scope.user.permissions = permissions;
     });
 
-    $scope.user.roles = [];
-    Permissions.getUserRoles($scope.user.username)
-    .success(function(response) {
-      $scope.user.roles = response.roles;
-    })
-    .error(function(response) {
-      if (typeof response === 'object') {
-        $scope.errors = response;
-        sweetAlert(
-          "Failed to load User Roles",
-          "error"
-        );
-      } else if (typeof response === 'string') {
-        sweetAlert(
-          "Failed to load User Roles" +
-          "(" + response+ ")",
-          "error"
-        );
-      }
-    })
-    .finally(function() {
+    $scope.getUserRoles = function (username){
+      $scope.users.forEach(function(eachUser){
+        if(eachUser.username === username){
+          $scope.user.roles = eachUser.roles;
+        }
+      });
```

Comment:
There's no reason to define getUserRoles() only to call it immediately afterwards -- just drop the function and run for the forEach directly.

Reply:
Oh yeah. I will take it out. Thanks